File: src/main/java/github/jk1/smtpidea/server/base/MessageFolder.java
package github.jk1.smtpidea.server.base;

import java.util.Collection;

/**
 * @author Evgeny Naumenko
 */
public interface MessageFolder {

    /**
     *
     * @return - collection of all messages in this folder
     */
    public Collection<MessageResource> getMessages();

    /**
     *
     * @return -  number of messages in this folder
     */
    public int numMessages();

    /**
     *
     * @return - sum of octet size of all messages in this folder
     */
    public int totalSize();
}

File: src/main/kotlin/github/jk1/smtpidea/store/MessageFolder.kt
import javax.swing.table.AbstractTableModel
import github.jk1.smtpidea.store.Clearable

/**
 *
 */
public abstract class MessageFolder<T> : AbstractTableModel(), Clearable {

    /**
     * @return collection of all messages in this folder
     */
    public abstract fun getMessages(): List<T>

    /**
     * @return number of messages in this folder
     */
    public abstract fun messageCount(): Int

    /**
     * @return a particular message by it's number
     */
    public abstract fun get(i: Int): T


    public abstract fun add(message: T)
}
